rpc/wrtc_base_channel.go

```diff
@@ -75,15 +75,7 @@ func newBaseChannel(
 		}
 	}
 	connStateChanged := func(connectionState webrtc.ICEConnectionState) {
-		ch.mu.Lock()
```

Comment:
To be honest, I'm not sure why I ever wrote this part of the code in. It's deleted because now with an ICE connection closing down due to pc.Close, it may fire off its connection state changing which in turn calls this, causing a deadlock. I don't see why we should keep it!

Reply:
I suspect before with `peerConn.Close` not blocking, we could get this race/leak:
```
| connStateChanged                                | webrtcBaseChannel.Close                             |
|-------------------------------------------------+-----------------------------------------------------|
|                                                 | closeWithReason                                     |
|                                                 | - Lock                                              |
|                                                 | - closed = true                                     |
|                                                 | - PeerConn.Close                                    |
|                                                 | - Unlock                                            |
| activeBackgroundWorkers.Add(1) (0 workers -> 1) | activeBackgroundWorkers.Wait (0 workers -> returns) |
| Spawn Goroutine                                 |                                                     |
```

And now I suspect with close blocking -- it's guaranteed that either:
* We'll spin up the onStateChange callback + goroutine before* Close gets called or
* Close returning guarantees `connStateChanged` can no longer be invoked.

Reply:
Yeah I think you're spot on